langsecure/trace.py

```diff
@@ -0,0 +1,97 @@
+import threading
+from pathlib import Path
+from typing import IO, Sequence
+from opentelemetry import trace
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from opentelemetry.sdk.trace import ReadableSpan
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+from langfuse import Langfuse
+import json
+
+trace.set_tracer_provider(TracerProvider())
+processed_spans = {} # to avoid duplicate spans during export because of threading?
+
+class FileSpanExporter(SpanExporter):
+    def __init__(
+        self,
+        file_path: str | Path | IO,
+    ) -> None:
+        self.file_path = Path(file_path) if isinstance(file_path, str) else file_path
+        self._lock = threading.Lock()
+        self._file: IO | None = None
+
+    def export(self, spans: Sequence[ReadableSpan]) -> SpanExportResult:
+        global processed_spans
```

Comment:
this will keep growing with the spans ?
In case of server mode this could be a problem ?
Can the entries in this be deleted once the span is written into file ?

Reply:
exporter does not maintain list of spans (it is input to export) so we should not delete. I think opentelemetry would be managing it so it should not grow too big (that would be issue with other exporters as well).

In server mode, do we plan to use file base tracing? Maybe using central lanffuse would be better. We may need to log additional metadata in server mode to differentiate different client requests.